scalafix-core/shared/src/main/scala/scalafix/patch/Patch.scala

```diff
@@ -198,6 +198,13 @@ object Patch {
     patch.isEmpty || hasLintMessage && onlyLint
   }
 
+  private[scalafix] def isAllTokenPatchAtomic(patch: Patch): Boolean =
+    patch match {
+      case Concat(a, b) => isAllTokenPatchAtomic(a) && isAllTokenPatchAtomic(b)
+      case _: TokenPatch => false
```

Comment:
```scala
case _: AtomicPatch|  _: LintPatch  => true
case _ => false
```
?

Reply:
@olafurpg it would return `false` for `EmptyPatch` and `TreePatch` (imports + symbols). Please see my comments: https://github.com/scalacenter/scalafix/pull/720#discussion_r186379813 / https://github.com/scalacenter/scalafix/pull/720#discussion_r185717259 . wdyt?

Reply:
It should return `true` for `EmptyPatch` but `false` for `TreePatch`, Tree patches are intended to be combined into larger atomic units.

Reply:
```scala
case _: AtomicPatch |  _: LintPatch | _: EmptyPatch => true
case _ => false
```
?

Reply:
Correct @joan38